File: bitManipulation/136/BetterSolution.php
<?php

class BetterSolution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function singleNumber($nums) {
        $res = 0;
        for($i=0; $i < count($nums); $i++){
            $res ^= $nums[$i];
        }
        return $res;
    }
}

File: array/238/BetterSolution.py
class BetterSolution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        length = len(nums)
        res = [1] * length
        prefix = 1
        for i in range(length):
            res[i] = prefix
            prefix *= nums[i]
        postfix=1
        for i in range(length-1, -1, -1):
            res[i] *= postfix
            postfix *= nums[i]
        return res

